### PR TITLE
Promo component

### PR DESCRIPTION
**Pull request by gestchild**
## What is this PR trying to achieve?

- Makes the promo component more generic for use in the various explore page blocks.
- Adds some styles for the different promo variants.

## What does it look like?
screencapture-localhost-3000-components-preview-promo-1484832440963

## For interface components
- [X] Browser testing - Have you checked the component in our supported browsers
- [X] No javascript - Have you checked the component with javascript disabled
- [ ] A11y testing - Have you run pa11y against the component and fixed/verified all errors, warnings and notices?

### CODE REVIEW

File: client/scss/components/_promo.scss [hunk 1]
```diff
@@ -13,13 +11,55 @@
...
+}
+
+.promo--video {
+  .icon { // aspect ratio: 17 x 19
+    width: (17/19) * 1.6em;
+    height: (19/19) * 1.6em;
+  }
+}
+
+.promo--underlined {
```
Comment (davidpmccormick) [inline client/scss/components/_promo.scss]: Only prob I had with this was that the underline wasn't on the promo when it was in a `.grid__cell--l12`.

File: client/scss/components/_promo.scss [hunk 2]
```diff
@@ -28,10 +68,102 @@
...
+
+.promo__clip-path--8 {
+  clip-path: polygon(0% 45%, 0% 100%, 100% 100%, 100% 45%, 82% 52%, 74% 44%, 60% 50%, 55% 57%, 40% 52%, 27% 48%);
+}
+
+.promo__clip-path--9 {
+  clip-path: polygon(0% 65%, 0% 100%, 100% 100%, 100% 30%, 84% 37%, 69% 32%, 60% 60%, 55% 63%, 40% 60%, 27% 64%);
+}
+
+.promo__clip-path--10 {
```
Comment (davidpmccormick) [inline client/scss/components/_promo.scss]: These are cool! Is there scope for generating them randomly in the future?

File: server/views/components/promo/index.njk [hunk 3]
```diff
@@ -1,14 +1,64 @@
...
+        <svg class="promo__defs">{# TODO proper filter - see Tomi #}
+          <filter id="duotone" color-interpolation-filters="sRGB" x="0" y="0" height="100%" width="100%">
+            <feColorMatrix type="matrix"
+              values="0.90 0 0 0   0.40 
+                      0.95 0 0 0  -0.10  
+                    -0.20 0 0 0   0.65 
+                        0  0 0 1   0" />
+          </filter>
+        </svg>
+        <img class="promo__image-mask promo__clip-path--{{ [1,2,3,4,5,6,7,8,9,10] | random }}" src="{{image.sources[(image.sources | length) - 1].src}}" alt="" />
```
Comment (davidpmccormick) [inline server/views/components/promo/index.njk]: Nice.

File: server/views/components/promo/index.njk [hunk 4]
```diff
@@ -1,14 +1,64 @@
...
+{#TODO styling video/podcast/gallery version#}
+<a href="{{ url }}" class="promo  {% if meta.type == 'gallery' or meta.type == 'podcast' or meta.type == 'video' %}promo--{{meta.type}}{% endif %} {% for modifier in modifiers %}promo--{{ modifier }} {% endfor %} ">
+  <div class="promo__image-container">
+    <img class="promo__image"
+      src="{{ image.sources[(image.sources | length) - 1].src }}"
+        srcset="{%- for source in image.sources %}
+          {{ source.src }} {{ source.size }}w
+          {%- if (source !== (image.sources | last)) %}, {%- endif %}
+        {%- endfor %}"
+        alt="" />
```
Comment (davidpmccormick) [inline server/views/components/promo/index.njk]: Keep `{{ image.alt }}` in this one?

File: server/views/components/promo/index.njk [hunk 5]
```diff
@@ -1,14 +1,64 @@
...
+                        0  0 0 1   0" />
+          </filter>
+        </svg>
+        <img class="promo__image-mask promo__clip-path--{{ [1,2,3,4,5,6,7,8,9,10] | random }}" src="{{image.sources[(image.sources | length) - 1].src}}" alt="" />
+      {% endif %}
+    {% endfor %}
+
+    {% if meta.type == 'gallery' or meta.type == 'podcast' or meta.type == 'video' %}
+      <div class="promo__icon-container">
+        {% if meta.type == 'gallery' %}
```
Comment (davidpmccormick) [inline server/views/components/promo/index.njk]: Would `{% icon meta.type %}` work here?

Comment (davidpmccormick) [inline server/views/components/promo/index.njk]: I think I've seen `===` as the equality checker elsewhere – not sure if it's important here.

File: server/views/components/promo/index.njk [hunk 6]
```diff
@@ -1,14 +1,64 @@
...
+          {% icon gallery %}
+        {% elif meta.type == 'podcast' %}
+          {% icon podcast %}
+        {% elif meta.type == 'video' %}
+            {% icon video %}
+        {% endif %}
+        {% if meta.type == 'gallery' and meta.length %}
+          <span class="promo__length"><span class="promo__sr">Gallery with</span> {{ meta.length }} <span class="promo__sr">items</span></span>
```
Comment (davidpmccormick) [inline server/views/components/promo/index.njk]: In the admittedly unlikely event of having a gallery with one item, would `item{{ 's' if meta.length !== 1 }}` be safer? (not sure if that expression works).

File: server/views/components/promo/index.njk [hunk 7]
```diff
@@ -1,14 +1,64 @@
...
+        {% endif %}
+      </div>
+    {% endif %}
+  </div>
   <header class="promo__description">
-    <span class="promo__intro">{{ intro }}</span>
-    <h2 class="promo__title"><a href="{{ url }}" class="promo__link">{{ title }}</a></h2>
-    <span class="promo__copy">{{ copy }}</span>
+    {% if meta.type %}
+      <span class="promo__type" aria-hidden="true">{{ meta.type }}</span>
```
Comment (davidpmccormick) [inline server/views/components/promo/index.njk]: Why does this guy get ARIA hidden?

**Comment by jamesgorrie** [comment]
Can we please please define what we mean by Promo?
It sounds like an ad?

File: client/scss/components/_promo.scss [hunk 8]
```diff
@@ -13,13 +11,55 @@
...
+  border-bottom: 2px solid color('black');
+}
+
+.promo__image-container {
+  position: relative;
   width: 100%;
   display: block;
   margin-bottom: 1rem;
   border-radius: $border-radius-unit;
+  overflow: hidden;
```
Comment (jamesgorrie) [inline client/scss/components/_promo.scss]: Hmmm... What's this for?

File: client/scss/components/_promo.scss [hunk 9]
```diff
@@ -28,10 +68,102 @@
   }
 }
 
-.promo__intro {
+.promo__image {
+  display: block;
+  width: 100%;
+}
+
+.promo__defs {
```
Comment (jamesgorrie) [inline client/scss/components/_promo.scss]: `defs`?

File: client/scss/components/_promo.scss (hunk 2 above)
Comment (jamesgorrie) [inline client/scss/components/_promo.scss]: I'm worried about the browser perf on all of this?
We are SVGing the hell out of the site... Any scope to get a low perf device for the team @jennpb?

File: client/scss/components/_promo.scss [hunk 10]
```diff
@@ -28,10 +68,102 @@
...
+  @include respond-to('large') {
+    padding: map-get($gutter-width, 'large')/2;
+  }
+}
+
+.promo__length {
+  display: inline-block;
+}
+
+.promo__sr {
```
Comment (jamesgorrie) [inline client/scss/components/_promo.scss]: What is `sr`?

File: server/views/components/promo/index.njk [hunk 11]
```diff
@@ -1,14 +1,64 @@
-<div class="promo">
-  <img class="promo__image"
-    src="{{ image.sources[(image.sources | length) - 1].src }}"
-      srcset="{%- for source in image.sources %}
-        {{ source.src }} {{ source.size }}w
-        {%- if (source !== (image.sources | last)) %}, {%- endif %}
-      {%- endfor %}"
-      alt="{{ image.alt }}" />
+{% set gallery = 'actions/gallery-link' %}
```
Comment (jamesgorrie) [inline server/views/components/promo/index.njk]: Why not just have these in the code?
Ot name them `galleryIcon` to make more sense?

File: server/views/components/promo/index.njk [hunk 12]
```diff
@@ -1,14 +1,64 @@
...
+{% set podcast = 'actions/volume' %}
+{% set video = 'actions/play' %}
+{#TODO full width variant?#}
+{#TODO full width/dark background variant?#}
+{#TODO Test with screen reader?#}
+{#TODO styling video/podcast/gallery version#}
+<a href="{{ url }}" class="promo  {% if meta.type == 'gallery' or meta.type == 'podcast' or meta.type == 'video' %}promo--{{meta.type}}{% endif %} {% for modifier in modifiers %}promo--{{ modifier }} {% endfor %} ">
```
Comment (jamesgorrie) [inline server/views/components/promo/index.njk]: Let's definitely take a look at writing a className utility for this.

File: server/views/components/promo/index.config.yml [hunk 13]
```diff
@@ -1,15 +1,132 @@
...
+        size: '900'
+      - src: https://placehold.it/450x300
+        size: '450'
+      alt: placeholder image
+    meta: 
+      type: video
+      date:
+      length: '1:37'
+    title: Tomorrow Belongs to Me
+    copy:
```
Comment (jamesgorrie) [inline server/views/components/promo/index.config.yml]: This is a great example of where using a [JS config](https://github.com/wellcometrust/wellcomecollection.org/pull/355/files#diff-00b52530f56131febb32f260c08eb872R1) would be useful.

File: server/views/components/promo/index.njk (hunk 5 above)
Comment (jamesgorrie) [inline server/views/components/promo/index.njk]: `{% icon 'actions/'+ meta.type %}` Would.

Comment (jamesgorrie) [inline server/views/components/promo/index.njk]: `===` is what we've been using as it's still JS.

File: server/views/components/promo/index.njk (hunk 6 above)
Comment (jamesgorrie) [inline server/views/components/promo/index.njk]: Nice! That's the wording I was looking for in the gallery.

File: server/views/components/promo/index.njk [hunk 14]
```diff
@@ -1,14 +1,64 @@
...
+      src="{{ image.sources[(image.sources | length) - 1].src }}"
+        srcset="{%- for source in image.sources %}
+          {{ source.src }} {{ source.size }}w
+          {%- if (source !== (image.sources | last)) %}, {%- endif %}
+        {%- endfor %}"
+        alt="" />
+
+    {% for modifier in modifiers %}
+      {% if modifier == 'series' %}
+        <svg class="promo__defs">{# TODO proper filter - see Tomi #}
```
Comment (jamesgorrie) [inline server/views/components/promo/index.njk]: Still unsure as to what this is.

File: client/scss/components/_promo.scss (hunk 1 above)
Comment (gestchild) [inline client/scss/components/_promo.scss]: I can add a container query

File: server/views/components/promo/index.njk [hunk 15]
```diff
@@ -1,14 +1,64 @@
...
+{% set gallery = 'actions/gallery-link' %}
+{% set podcast = 'actions/volume' %}
+{% set video = 'actions/play' %}
+{#TODO full width variant?#}
+{#TODO full width/dark background variant?#}
+{#TODO Test with screen reader?#}
+{#TODO styling video/podcast/gallery version#}
+<a href="{{ url }}" class="promo  {% if meta.type == 'gallery' or meta.type == 'podcast' or meta.type == 'video' %}promo--{{meta.type}}{% endif %} {% for modifier in modifiers %}promo--{{ modifier }} {% endfor %} ">
+  <div class="promo__image-container">
+    <img class="promo__image"
```
Comment (jamesgorrie) [inline server/views/components/promo/index.njk]: Once we get the new image component in we can generify all of this.

File: client/scss/components/_promo.scss (hunk 2 above)
Comment (gestchild) [inline client/scss/components/_promo.scss]: Don't see why not, might do it server side and inject as an inline style

File: server/views/components/promo/index.njk (hunk 4 above)
Comment (gestchild) [inline server/views/components/promo/index.njk]: The image is inside the link now, so thought this was better empty

File: server/views/components/promo/index.njk (hunk 5 above)
Comment (gestchild) [inline server/views/components/promo/index.njk]: The icon takes the icon folder/filename, which is different to the type

File: server/views/components/promo/index.njk (hunk 6 above)
Comment (gestchild) [inline server/views/components/promo/index.njk]: I'll see if it does

File: server/views/components/promo/index.njk (hunk 7 above)
Comment (gestchild) [inline server/views/components/promo/index.njk]: The text above already says 'Gallery/Video/Podcast' so didn't want to repeat it here, although it should be read for the articles. I'll put in a conditional

File: client/scss/components/_promo.scss (hunk 9 above)
Comment (gestchild) [inline client/scss/components/_promo.scss]: Relates to the svg filter, used on the --series promo

https://developer.mozilla.org/en-US/docs/Web/SVG/Element/defs

File: client/scss/components/_promo.scss (hunk 8 above)
Comment (gestchild) [inline client/scss/components/_promo.scss]: stops the image corners showing under the border-radius

File: server/views/components/promo/index.njk (hunk 5 above)
Comment (davidpmccormick) [inline server/views/components/promo/index.njk]: > The icon takes the icon folder/filename, which is different to the meta.type

Oh yeah – ignore me.

File: client/scss/components/_promo.scss (hunk 9 above)
Comment (jamesgorrie) [inline client/scss/components/_promo.scss]: My lack of knowledge...

File: client/scss/components/_promo.scss (hunk 2 above)
Comment (jennpb) [inline client/scss/components/_promo.scss]: @jamesgorrie I can find out...what do you suggest?

**Pull request merged**